cpp/src/barretenberg/dsl/acir_format/ecdsa_secp256k1.cpp

```diff
@@ -84,9 +84,15 @@ witness_ct ecdsa_index_to_witness(Composer& composer, uint32_t index)
     return { &composer, value };
 }
 
-void create_ecdsa_verify_constraints(Composer& composer, const EcdsaSecp256k1Constraint& input)
+void create_ecdsa_verify_constraints(Composer& composer,
+                                     const EcdsaSecp256k1Constraint& input,
+                                     bool has_valid_witness_assignments)
```

Comment:
This is a `nit` but you could set a default to `false` and avoid having to write out the param for `create_ecdsa_verify_constraints` when there is not a valid witness assignment

Reply:
That's a good point, I think we want to make the default = true though so that it fails fast and not replace witness values by default.